x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/create_blocklist_form.tsx

```diff
@@ -0,0 +1,472 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState, useCallback, memo } from 'react';
+import styled from 'styled-components';
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiTextArea,
+  EuiHorizontalRule,
+  EuiText,
+  EuiSpacer,
+  EuiSuperSelect,
+  EuiSuperSelectOption,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import {
+  OperatingSystem,
+  ConditionEntryField,
+  isPathValid,
+  hasSimpleExecutableName,
+} from '@kbn/securitysolution-utils';
+import { isOneOfOperator } from '@kbn/securitysolution-list-utils';
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+
+import { OS_TITLES } from '../../../../common/translations';
+import { ArtifactFormComponentProps } from '../../../../components/artifact_list_page';
+import {
+  CONDITIONS_HEADER,
+  CONDITIONS_HEADER_DESCRIPTION,
+  CONDITION_FIELD_DESCRIPTION,
+  CONDITION_FIELD_TITLE,
+  DESCRIPTION_LABEL,
+  DETAILS_HEADER,
+  DETAILS_HEADER_DESCRIPTION,
+  FIELD_LABEL,
+  NAME_LABEL,
+  OPERATOR_LABEL,
+  POLICY_SELECT_DESCRIPTION,
+  SELECT_OS_LABEL,
+  VALUE_LABEL,
+  ERRORS,
+} from '../../translations';
+import {
+  EffectedPolicySelect,
+  EffectedPolicySelection,
+} from '../../../../components/effected_policy_select';
+import {
+  GLOBAL_ARTIFACT_TAG,
+  BY_POLICY_ARTIFACT_TAG_PREFIX,
+} from '../../../../../../common/endpoint/service/artifacts/constants';
+import { useLicense } from '../../../../../common/hooks/use_license';
+import { isValidHash } from '../../../../../../common/endpoint/service/trusted_apps/validations';
+import { isArtifactGlobal } from '../../../../../../common/endpoint/service/artifacts';
+import type { PolicyData } from '../../../../../../common/endpoint/types';
+
+interface BlocklistEntry {
+  field: ConditionEntryField;
+  operator: 'included';
+  type: 'match_any';
+  value: string[];
+}
+
+interface ItemValidation {
+  name?: React.ReactNode[];
+  value?: React.ReactNode[];
+}
+
+function createValidationMessage(message: string): React.ReactNode {
+  return <div>{message}</div>;
+}
+
+function getDropdownDisplay(field: ConditionEntryField): React.ReactNode {
+  return (
+    <>
+      {CONDITION_FIELD_TITLE[field]}
+      <EuiText size="xs" color="subdued">
+        {CONDITION_FIELD_DESCRIPTION[field]}
+      </EuiText>
+    </>
+  );
+}
+
+const InputGroup = styled.div`
```

Comment:
We already have this implementation [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/condition_entry_input/index.tsx), maybe is not needed now, but would be cool having this `condition_entry_input` in a higher level and be generic.

Reply:
I did contemplate using `ConditionEntryInput` / `ExceptionBuilderComponent` instead but there were enough minor differences that upgrading the existing components to support the requirements here were a bit too time consuming for the scope.